torch/csrc/jit/codegen/cuda/index_compute.cpp

```diff
@@ -1491,10 +1496,12 @@ std::vector<kir::Val*> Index::getGlobalProducerStridedIndices(
 }
 
 // Producer index for either shared or local memory
-std::vector<kir::Val*> Index::getNonGlobalProducerStridedIndices(
-    TensorView* producer_tv,
-    const TensorView* consumer_tv,
-    const std::vector<kir::ForLoop*>& loops) {
+std::tuple<std::vector<kir::Val*>, std::vector<kir::Val*>> Index::
```

Comment:
Only thing done was to provide an option in this function to return both the indices and strides vectors instead of the pointwise product of the two.

Reply:
I totally agree that a `index` + `linearize` re-formalize would make sense. But the part I wasn't sure is if it is useful anywhere other than for mma. It is a bit hard to justify re-formulating the indexing scheme here if keep_stride option is mma exclusive. 

I'm actually thinking about some indexing clean up in a follow up PR if that's ok. We probably needed to discuss this a bit before going for a formal approach and re-factor.

Reply:
> I would actually prefer if we linearize here by having a swizzle op on IterDomains. I think that would generally prevent needing separate linearization logic in this PR. It could also allow to have swizzle ops that are not applied directly on leaves, I don't know if this would be needed, but seems easy enough to do with this change.
> 
> I think the hardest part of doing this would be just figuring out which replays need to ignore swizzle, and which don't. That might take some careful changing, but don't think it'd be too bad.
> 
> Would be nice to also change the transpose swizzle to behave like this.

I like the idea. If we do so, I think it's good to have a separate PR just adding the swizzle interface.

Reply:
(This thread seems out of sync) Yes I think we want to build everything into index compute eventually by defining a swizzle op for iterdomains, like in this comment:
[https://github.com/csarofeen/pytorch/pull/1274/files/bb46d04b72a31b91a14aff1604aeabc5ed86f2a5#r755424473](https://github.com/csarofeen/pytorch/pull/1274/files/bb46d04b72a31b91a14aff1604aeabc5ed86f2a5#r755424473)

That doesn't look like a small feature to build though. I would prefer we could build it in parallel in a separate PR and come back to do an cleanup here.